app/src/androidTest/java/comcmput301f16t01/github/carrier/ElasticUserControllerTest.java

```diff
@@ -0,0 +1,72 @@
+package comcmput301f16t01.github.carrier;
+
+import java.util.Random;
+
+/**
+ * Created by kiete on 11/13/2016.
+ */
+
+import java.util.Random;
+import java.util.concurrent.ExecutionException;
+
+public class ElasticUserControllerTest extends ApplicationTest {
+    UserController uc = new UserController();
+
+    @Override
+    protected void setUp() throws InterruptedException {
+        // The logged in user
+        uc.createNewUser("ElasticUserControllerTest", "test@test.com", "1234567890");
+        Thread.sleep(1000);
+        uc.logInUser("ElasticUserControllerTest");
+        // A user to remove later
+        uc.createNewUser("ElasticUserControllerTest2", "delete@test.com", "1234567890");
+        Thread.sleep(1000);
+    }
+
+    @Override
+    protected void tearDown() {
+        uc.deleteUser("ElasticUserControllerTest");
+    }
+
+    public void testAddUserTask() throws InterruptedException, ExecutionException {
+        User elasticUser = uc.findUser("ElasticUserControllerTest");
+        Thread.sleep(1000);
+
+        // Ensure that the user was actually put into elastic search
+        assertEquals("User in elastic search and copy of user are not the same",
+                UserController.getLoggedInUser().getUsername(), elasticUser.getUsername());
+
+
+    }
+
+    public void testUniqueUsername() {
+
+//        Test for uniqueness, createNewUser returns the string "That username is already taken" if
+//        a string is not unique
+        assertEquals("The username is not unique.", "That username is already taken!", UserController.createNewUser
```

Comment:
This test looks wrong to me. What are you comparing the new user with? A string?